File: practica2_839756_842236/c++/instruccionWrite.cc
// Autores: Andrei Vlasceanu [839756] & Andres Yubero [842236]

#include "instruccionWrite.h"

void InstruccionWrite::ejecutar(stack<int> *s)
{
    int x;
    x = s->top();
    s->pop();
    cout << x << endl;
}

void InstruccionWrite::listar()
{
    cout << "write" << endl;
}

File: practica2_839756_842236/java/InstruccionWrite.java
import java.util.Stack;

public class InstruccionWrite extends Instruccion {
    @Override
    public void ejecutar(Stack<Integer> s) {
        int x = s.pop();
        System.out.println(x);
    }

    @Override
    public void listar() {
        System.out.println("write");
    }
}
